analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldInGenericClass.cs

```diff
@@ -72,11 +68,28 @@ private static void CheckMember(SonarSyntaxNodeReportingContext context, SyntaxN
         }
     }
 
+    private static string[] CollectTypeParameterNames(SyntaxNode current)
+    {
+        var names = new HashSet<string>();
+        while (current is not null)
+        {
+            if (current is TypeDeclarationSyntax { TypeParameterList: not null } typeDeclaration)
+            {
+                names.AddRange(typeDeclaration.TypeParameterList.Parameters.Select(x => x.Identifier.ValueText));
+            }
+            current = current.Parent;
+        }
+        return names.ToArray();
+    }
+
     private static bool HasGenericType(SonarSyntaxNodeReportingContext context, SyntaxNode root, string[] typeParameterNames) =>
         root.DescendantNodesAndSelf()
             .OfType<IdentifierNameSyntax>()
             .Any(x => typeParameterNames.Contains(x.Identifier.Value) && context.SemanticModel.GetSymbolInfo(x).Symbol is { Kind: SymbolKind.TypeParameter });
 
+    private static bool BaseTypeHasGenericTypeArgument(TypeDeclarationSyntax typeDeclaration, string[] typeParameterNames) =>
+        typeDeclaration.BaseList is not null && typeDeclaration.BaseList.Types.Any(x => x.Type is GenericNameSyntax genericType && HasGenericTypeArgument(genericType, typeParameterNames));
```

Comment:
```suggestion
        typeDeclaration.BaseList is {} baselist && baseList.Types.Any(x => x.Type is GenericNameSyntax genericType && HasGenericTypeArgument(genericType, typeParameterNames));
```
I like this more - up to you.